kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java

```diff
@@ -359,13 +382,18 @@ public synchronized void apply(String deviceId) throws DBusException {
 
     private synchronized void doApply(Map<String, Object> networkConfiguration) throws DBusException {
         logger.info("Applying configuration using NetworkManager Dbus connector");
-        List<Device> availableDevices = this.networkManager.getAllDevices();
+        NetworkProperties properties = new NetworkProperties(networkConfiguration);
+        List<String> availableDeviceIds = getInterfaceIds();
+        Set<String> availableDevices = new LinkedHashSet<String>(availableDeviceIds);
+        Optional<List<String>> configuredInterfaceIds = properties.getOptStringList("net.interfaces");
+        if (configuredInterfaceIds.isPresent()) {
+            availableDevices.addAll(configuredInterfaceIds.get());
+        }
```

Comment:
I think this can be simplified since `net.interface` property is not optional.

```suggestion
        NetworkProperties properties = new NetworkProperties(networkConfiguration);
        List<String> availableDeviceIds = getInterfaceIds();
        List<String>> configuredInterfaceIds = properties.getStringList("net.interfaces");

        Set<String> availableDevices = new LinkedHashSet<String>(availableDeviceIds);
        availableDevices.addAll(configuredInterfaceIds);
```

Reply:
I agree, but it breaks this test
https://github.com/eclipse/kura/blob/f99573b3fe5efd6ff7ea2411e54982f33d3cce8e/kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java#L327-L341